core/src/client/importer.rs

```diff
@@ -365,14 +365,19 @@ impl Importer {
     /// This is triggered by a message coming from a header queue when the header is ready for insertion
     pub fn import_verified_headers(&self, client: &Client) -> usize {
         let max_headers_to_import = 256;
+        let lock = self.import_lock.lock();
+        let headers = self.header_queue.drain(max_headers_to_import);
+        self.import_headers(headers, client, &lock)
+    }
 
-        let _lock = self.import_lock.lock();
+    fn import_headers(&self, headers: Vec<Header>, client: &Client, _importer_lock: &MutexGuard<()>) -> usize {
```

Comment:
I found no reason to move headers.
How about changing it to take `&[Header]`?

Reply:
```rust
pub fn import_headers<'a>(&'a self, headers: impl IntoIterator<Item = &'a Header>, client: &Client, _importer_lock: &MutexGuard<()>) -> usize {
```
would be better.
It can remove the cloning of header in https://github.com/CodeChain-io/codechain/pull/1481/commits/a17aec5dd566b6289a4d5b397c848d39465337fc